### DISCUSSION TITLE
GraphQL from Custom API

### DISCUSSION THREAD

**Discussion by marcusjm** [OrchardCMS/OrchardCore#11011] [Q&A] [answered]
I have done this with Chocolate but not sure how to best do it here. If iI have my own Web API with various CQRS methods returning data. How would I best expose them in the GraphQL interface? The data is not CMS data but various data from a custom system.

Or is there some extension where I can choose WebAPI as query source instead of the typical Luscene?

Thanx.

**Comment by Skrypt** [comment, marked as answer]
Normally with a Lucene or SQL Query you can define a Schema for the GraphQL API; so you can also map custom database tables that way.
Else, yes, you will need to add some extensions to your GraphQL module that exposes your custom API methods.

**Comment by marcusjm** [comment]
Ah. So I can basically do the same way I do with the regular GraphQL Queries?

**Reply by Skrypt** [reply]
Since you want to UPDATE records I recommend a custom extension of GraphQL because we don't provide any mutations yet with the GraphQL endpoint. Also, you can use the REST Content API to do that.

**Reply by marcusjm** [reply]
Ok I will try that. Thank You.
